lib/suse/connect/connection.rb

```diff
@@ -76,13 +77,15 @@ def json_request(method, path, params = {})
           request = VERB_TO_CLASS[method].new(path_with_params(path, params))
         else
           request = VERB_TO_CLASS[method].new(path)
-          request.body = params.to_json unless params.empty?
```

Comment:
I'm not 100% sure why it happened since I was not able to reproduce the bug from this branch. My thinking is that the package from the development repository had the bug from a previous stage of this branch that got fixed meanwhile, but I don't really recall doing something specific on this after pushing into obs.

Anyways, I've put extra `nil?` checks whenever we were calling `empty?` and a `nil` value could be given just to be more safe.